coreneuron/io/reports/report_event.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include <algorithm>
+#include <map>
+#include <vector>
+#include "coreneuron/network/netcon.hpp"
+#include "coreneuron/network/netcvode.hpp"
+
+namespace coreneuron {
+
+#if defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+struct VarWithMapping {
+    int id;
+    double* var_value;
+    VarWithMapping(int id_, double* v_) : id(id_), var_value(v_) {
+    }
+};
+
+// mapping the set of variables pointers to report to its gid
+typedef std::map<int, std::vector<VarWithMapping> > VarsToReport;
```

Comment:
isn't a `std::unordered_map` enough?
Besides I would use the `using` keyword to declare `VarsToReport`. Something like:

```suggestion
using VarsToReport =std::unordered_map<int, std::vector<VarWithMapping>>;
```